test/gtest/gpu_mha_backward.cpp

```diff
@@ -78,63 +80,62 @@ struct TestCase
     float dropout;
 };
 
-inline std::vector<TestCase> GetSmokeTestCases()
+inline std::vector<TestCase> GetSmokeCases()
 {
-    if(CheckFloatArg("--float"))
-    {
-        return {
-            {9, 8, 8, 8, 0.0f},
-            {1, 2, 4, 5, 0.0f},
-            {2, 1, 5, 4, 0.0f},
-            {4, 2, 1, 3, 0.0f},
-            {5, 3, 4, 1, 0.0f},
-            {1, 2, 65, 5, 0.0f},
-            {2, 1, 67, 4, 0.0f},
-            {8, 7, 68, 1, 0.0f},
-            {1, 2, 257, 5, 0.0f},
-            {2, 1, 259, 4, 0.0f},
-            {8, 7, 270, 1, 0.0f},
-            {1, 1, 1, 1, 0.0f},
-            {3, 5, 32, 7, 0.8f},
-            {2, 2, 64, 128, 0.8f},
-            {2, 1, 128, 4, 0.8f},
-            {2, 7, 256, 31, 0.8f},
-        };
-    }
-    else
+    if(!(CheckFloatArg("--float") || CheckFloatArg("--float8")))
     {
         return {};
     }
+
+    return {
+        {9, 8, 8, 8, 0.0f},
+        {1, 2, 4, 5, 0.0f},
+        {2, 1, 5, 4, 0.0f},
+        {4, 2, 1, 3, 0.0f},
+        {5, 3, 4, 1, 0.0f},
+        {1, 2, 65, 5, 0.0f},
+        {2, 1, 67, 4, 0.0f},
+        {8, 7, 68, 1, 0.0f},
+        {1, 2, 257, 5, 0.0f},
+        {2, 1, 259, 4, 0.0f},
+        {8, 7, 270, 1, 0.0f},
+        {1, 1, 1, 1, 0.0f},
+        {3, 5, 32, 7, 0.8f},
+        {2, 2, 64, 128, 0.8f},
+        {2, 1, 128, 4, 0.8f},
+        {2, 7, 256, 31, 0.8f},
+    };
 }
 
 inline std::vector<TestCase> GetFullTestCases()
 {
-    if(miopen::IsDisabled(ENV(MIOPEN_TEST_ALL)))
-        return {};
-
-    if(CheckFloatArg("--float"))
-    {
-        return {
-            {3, 15, 2047, 15, 0.0f},
-            {2049, 17, 7, 7, 0.0f},
-            {3, 3, 257, 91, 0.0f},
-            {11, 150, 255, 31, 0.0f},
-            {9, 3, 129, 1023, 0.0f},
-            {3, 15, 31, 2047, 0.0f},
-            {2049, 17, 32, 7, 0.2f},
-            {11, 150, 256, 31, 0.4f},
-        };
-    }
-    else
+    if(miopen::IsDisabled(ENV(MIOPEN_TEST_ALL)) ||
+       !(CheckFloatArg("--float") || CheckFloatArg("--float8")))
     {
         return {};
     }
+
+    return {
+        {3, 15, 2047, 15, 0.0f},
+        {2049, 17, 7, 7, 0.0f},
+        {3, 3, 257, 91, 0.0f},
+        {11, 150, 255, 31, 0.0f},
+        {9, 3, 129, 1023, 0.0f},
+        {3, 15, 31, 2047, 0.0f},
+        {2049, 17, 32, 7, 0.2f},
+        {11, 150, 256, 31, 0.4f},
+    };
 }
 } // namespace
 
+template <typename T>
 class Test_Bwd_Mha : public testing::TestWithParam<TestCase>
 {
+    static_assert(std::is_same_v<T, float> || std::is_same_v<T, float8>);
+
 protected:
+    using dO_T = std::conditional_t<std::is_same_v<T, float>, float, bfloat8>;
```

Comment:
so `dO_T` can only be `bfloat8` and not `float8`?

Reply:
Correct.
It is by cudnn documentation, and also we can save some workspace size, since FP32xBF8 gemm is possible while FP32xF8 is not, and we need additional buffer to make a conversion.
Right now we do it in-place.